File: data_visual.py
import numpy as np


method_set = {'baseline','cdir','conv','motorz'};
metric = 'precision'
datadict = {}
for method in method_set:
    file_patrn = "run_{}_eval,tag_eval_{}.csv".format(method,metric)
    fd = open(file_patrn,'r')
    d = np.loadtxt(fd,
           skiprows=1,
           delimiter=',',
           dtype={'names': ('time', 'step', 'value'),
           'formats': ('float', 'i4', 'float')}) #s4 for string
    d_list = [np.asarray([d_[1],d_[2]]) for d_ in d]
    d_matrix = np.concatenate(d_list,axis=0)
    datadict[method] = d_matrix
    fd.close()

File: data_visual.m
Precision = 0;
if Precision
    load('data_p.mat')
    method_set = {'baseline','cdir','motorz','conv'};

    figure();

    for i_count = 1:4
        method = method_set{i_count}
        data_var = [method '_p'];
        data = eval(data_var);
        x = data{1};
        y = data{2};    

        plot(x,y,'LineWidth',1.5)

        hold on;
    end
    xlim([0,600000])
    xlabel('Step count')
    ylabel('Precision')
    grid on
    legend(method_set{1},method_set{2},method_set{3},method_set{4})
else
    load('data_r.mat')
    method_set = {'baseline','cdir','motorz','conv'};

    figure();

    for i_count = 1:4
        method = method_set{i_count}
        data_var = [method '_r'];
        data = eval(data_var);
        x = data{1};
        y = data{2};    

        plot(x,y,'LineWidth',1.5)

        hold on;
    end
    xlim([0,600000])
    xlabel('Step count')
    ylabel('Recall')
    grid on
    legend(method_set{1},method_set{2},method_set{3},method_set{4})
end